adobe/aem-core-cif-components
# CIF-1788 - Implement ComponentExporter for featured category list component

## Description

Add the required annotations to the `FeaturedCategoryList` Sling Model and implement additional methods.

## Related Issue

CIF-1788

## Motivation and Context

Expose server-side data to the client-side components

## How Has This Been Tested?

Unit tests and functional testing.




Steps for functional testing:
1/ Install AEM Venia
2/ Install the CIF Core Components from this branch
3/ Open the home page and add a Featured Categories component
4/ Configure the component by adding two or more categories, at least one having an "override asset"
5/ Open the page using the model selector, e.g. `http://localhost:4502/content/venia/us/en.model.json`
6/ Look for the `featuredcategories` component in the JSON output and check the following:
* The `categoryItems` entry is an array
* Each category item has an `identifier` entry with the `{entityType: CATEGORY}` and `{type:ID}`
* The path to the override asset is correctly set in the `assetPath` entry.
* The `:type` property is set to `core/cif/components/commerce/featuredcategorylist/v1/featuredcategorylist`
## Screenshots (if appropriate):

## Types of changes




## Checklist:
